examples/penumbra/penumbra_ibc_test.go

```diff
@@ -0,0 +1,660 @@
+package penumbra
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"cosmossdk.io/math"
+	transfertypes "github.com/cosmos/ibc-go/v8/modules/apps/transfer/types"
+	"github.com/strangelove-ventures/interchaintest/v8"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/penumbra"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/strangelove-ventures/interchaintest/v8/relayer"
+	"github.com/strangelove-ventures/interchaintest/v8/testreporter"
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+)
+
+// TestPenumbraToPenumbraIBC asserts that basic IBC functionality works between two Penumbra testnet networks.
+// Two instances of Penumbra will be spun up, the relayer will be configured, and an ics-20 token transfer will be
+// sent from chainA -> chainB successfully. At the end two more ics-20 transfers will be sent from chainA -> chainB,
+// these transfers will time out due to the timeout height and timeout timestamp being reached respectively.
+func TestPenumbraToPenumbraIBC(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping in short mode")
+	}
+
+	t.Parallel()
+	client, network := interchaintest.DockerSetup(t)
+
+	nv := 1
+	fn := 0
+
+	chains, err := interchaintest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*interchaintest.ChainSpec{
+		{
+			Name: "penumbra",
+			// Version: "v0.77.2,v0.37.5",
+			Version: "local,v0.37.5",
+			ChainConfig: ibc.ChainConfig{
+				ChainID: "penumbraA-0",
+			},
+			NumValidators: &nv,
+			NumFullNodes:  &fn,
+		},
+		{
+			Name: "penumbra",
+			// Version: "v0.77.2,v0.37.5",
+			Version: "local,v0.37.5",
+			ChainConfig: ibc.ChainConfig{
+				ChainID: "penumbraB-0",
+			},
+			NumValidators: &nv,
+			NumFullNodes:  &fn,
+		},
+	},
+	).Chains(t.Name())
+	require.NoError(t, err, "failed to get penumbra chains")
+	require.Len(t, chains, 2)
+
+	chainA := chains[0].(*penumbra.PenumbraChain)
+	chainB := chains[1].(*penumbra.PenumbraChain)
+
+	i := ibc.DockerImage{
+		Repository: "relayer",
+		Version:    "local",
```

Comment:
local - needs to be changed to a version / branch

Reply:
This is currently running on the `justin-proto-update` branch, those changes should probably be merged -- also I'm not sure when the last proto update occurred, it is a good idea to regenerate the Penumbra go proto bindings from the `v0.78.0` tag (shouldn't be any breaking changes, but good to have parity regardless)

Reply:
Agree, ref: https://github.com/cosmos/relayer/pull/1474#pullrequestreview-2142736657

Reply:
1f22780b15b93cf2d97af85a97012b6682c19fb4 updated